src/modules/consumer.js

```diff
@@ -78,15 +78,13 @@ function consume (queue, options, callback) {
       return true;
     });
   })
-  .catch(() => {
-    //in case of any error creating the channel, wait for some time and then try to reconnect again (to avoid overflow)
-    return utils.timeoutPromise(this.conn.config.timeout)
-      .then(() => {
-        return consume.call(this, queue, options, callback);
-      });
-  });
+  .catch(() =>
+    // in case of any error creating the channel, wait for some time and then try to reconnect again (to avoid overflow)
+    utils.timeoutPromise(this.conn.config.timeout)
+      .then(() => consume.call(this, queue, options, callback))
+  );
 }
 
-module.exports = function(conn) {
+module.exports = function (conn) {
```

Comment:
@Spring3 Since you are doing the refactor, feel free to use arrow function but make sure there are no problems. That was written in the context of badly written code there where it is very trick to mistake this in a function and an arrow function and it needs a concise cleanup and refactor because of cross function scopes of objects.